backends/tfhe-cuda-backend/cuda/src/device.cu

```diff
@@ -5,6 +5,9 @@
 uint32_t cuda_get_device() {
   int device;
   check_cuda_error(cudaGetDevice(&device));
+#ifdef USE_NVTOOLS
+  cudaStartProfile();
+#endif
```

Comment:
Is it ok to call this many times if there are multiple threads calling cuda_get_device? Perhaps doing this like the mempool activation thing would be better? Mempool is merged on main already.

Reply:
Ok let's have faith for once :slightly_smiling_face: